File: frontend/src/api/investmentLogic.ts
import http from './http'

export interface InvestmentLogicEntry {
  id: number
  technical_content: string | null
  fundamental_content: string | null
  message_content: string | null
  weight_technical: number
  weight_fundamental: number
  weight_message: number
  created_at: string
  updated_at: string
  extra_json: Record<string, unknown> | null
}

export interface InvestmentLogicPayload {
  technical_content?: string | null
  fundamental_content?: string | null
  message_content?: string | null
  weight_technical: number
  weight_fundamental: number
  weight_message: number
  extra_json?: Record<string, unknown> | null
}

export function fetchInvestmentLogicCurrent() {
  return http.get<{ entry: InvestmentLogicEntry | null }>('/investment-logic/current')
}

export function fetchInvestmentLogicEntries(order: 'created_desc' | 'created_asc' = 'created_desc') {
  return http.get<{ items: InvestmentLogicEntry[] }>('/investment-logic/entries', {
    params: { order },
  })
}

export function createInvestmentLogicEntry(data: InvestmentLogicPayload) {
  return http.post<InvestmentLogicEntry>('/investment-logic/entries', data)
}

export function updateInvestmentLogicEntry(id: number, data: InvestmentLogicPayload) {
  return http.put<InvestmentLogicEntry>(`/investment-logic/entries/${id}`, data)
}

export function deleteInvestmentLogicEntry(id: number) {
  return http.delete(`/investment-logic/entries/${id}`)
}

File: backend/app/schemas/investment_logic.py
"""投资逻辑 API 模型。"""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class InvestmentLogicEntryOut(BaseModel):
    """单条条目响应。"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    technical_content: str | None = None
    fundamental_content: str | None = None
    message_content: str | None = None
    weight_technical: int
    weight_fundamental: int
    weight_message: int
    created_at: datetime
    updated_at: datetime
    extra_json: dict[str, Any] | None = None


class InvestmentLogicCreateIn(BaseModel):
    """POST 新增。"""

    technical_content: str | None = None
    fundamental_content: str | None = None
    message_content: str | None = None
    weight_technical: int = Field(..., ge=0, le=100)
    weight_fundamental: int = Field(..., ge=0, le=100)
    weight_message: int = Field(..., ge=0, le=100)
    extra_json: dict[str, Any] | None = None


class InvestmentLogicCurrentOut(BaseModel):
    entry: InvestmentLogicEntryOut | None = None


class InvestmentLogicListOut(BaseModel):
    items: list[InvestmentLogicEntryOut]
